presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java

```diff
@@ -226,6 +227,54 @@ public void merge(TDigest other)
         }
     }
 
+    /**
+     * Get the approx truncated mean from the digest.
+     * The mean is computed as a weighted average of values between the upper and lower quantiles (inclusive)
+     * When the rank of a quantile is non-integer, a portion of the nearest rank's value is included in the mean
+     * according to its fraction within the quantile bound.
+     * <p>
+     * If the value in the digest needs to be converted before the mean is calculated,
+     * valueFunction can be used. Else, pass in the identity function.
+     */
+    public Double getTruncatedMean(double lowerQuantile, double upperQuantile)
+    {
+        double sum = 0;
+        double count = 0;
+        double mean = 0;
+
+        compress();
+
+        if (totalWeight == 0 || lowerQuantile >= upperQuantile) {
```

Comment:
I think a precondition check is better than `null`.  Also let's check that the quantiles are sane values.